web/src/features/moreCast2/util.test.ts

```diff
@@ -345,3 +352,222 @@ describe('fillRowsFromSavedDraft', () => {
+    const yesterdayForecastRow = buildForecastRowWithIndices(1, YESTERDAY, 438, 89, NaN)
+    let forecastRow = buildValidForecastRow(1, TODAY)
+    forecastRow = calculateFWIs(yesterdayForecastRow, forecastRow)
+    expect(forecastRow.ffmcCalcForecast).toBe(undefined)
+    expect(forecastRow.isiCalcForecast).toBe(undefined)
+    expect(forecastRow.fwiCalcForecast).toBe(undefined)
+  })
+  it("should not calculate dmc, bui, fwi if dmc in yesterday's actual row is NaN", () => {
+    const yesterdayForecastRow = buildForecastRowWithIndices(1, YESTERDAY, 438, NaN, 87)
+    let forecastRow = buildValidForecastRow(1, TODAY)
+    forecastRow = calculateFWIs(yesterdayForecastRow, forecastRow)
+    expect(forecastRow.dmcCalcForecast).toBe(undefined)
+    expect(forecastRow.buiCalcForecast).toBe(undefined)
+    expect(forecastRow.fwiCalcForecast).toBe(undefined)
+  })
+  it("sshould not calculate dmc, bui, fwi if dmc in yesterday's forecast row is NaN", () => {
+    const yesterdayForecastRow = buildForecastRowWithIndices(1, YESTERDAY, 438, NaN, 87)
+    let forecastRow = buildValidForecastRow(1, TODAY)
+    forecastRow = calculateFWIs(yesterdayForecastRow, forecastRow)
+    expect(forecastRow.dmcCalcForecast).toBe(undefined)
+    expect(forecastRow.buiCalcForecast).toBe(undefined)
+    expect(forecastRow.fwiCalcForecast).toBe(undefined)
+  })
+  it("should not calculate dc, bui, fwi if dmc in yesterday's actual row is NaN", () => {
+    const yesterdayForecastRow = buildForecastRowWithIndices(1, YESTERDAY, NaN, 89, 87)
+    let forecastRow = buildValidForecastRow(1, TODAY)
+    forecastRow = calculateFWIs(yesterdayForecastRow, forecastRow)
+    expect(forecastRow.dcCalcForecast).toBe(undefined)
+    expect(forecastRow.buiCalcForecast).toBe(undefined)
+    expect(forecastRow.fwiCalcForecast).toBe(undefined)
+  })
+  it("sshould not calculate dc, bui, fwi if dmc in yesterday's forecast row is NaN", () => {
+    const yesterdayForecastRow = buildForecastRowWithIndices(1, YESTERDAY, NaN, 89, 87)
+    let forecastRow = buildValidForecastRow(1, TODAY)
+    forecastRow = calculateFWIs(yesterdayForecastRow, forecastRow)
+    expect(forecastRow.dcCalcForecast).toBe(undefined)
+    expect(forecastRow.buiCalcForecast).toBe(undefined)
+    expect(forecastRow.fwiCalcForecast).toBe(undefined)
+  })
+})
+
+describe('simulateFireWeatherIndices', () => {
+  it('should return input unchanged if only one row present per station', () => {
+    const forecastRow = buildValidForecastRow(1, TODAY)
+    const result = simulateFireWeatherIndices([forecastRow])
+    expect(result[0]).toBe(forecastRow)
+  })
+  it('should simulate FWIs for all forecast rows', () => {
+    const forecastRowA = buildForecastRowWithIndices(1, YESTERDAY, 438, 89, 87)
+    const forecastRowB = buildValidForecastRow(1, TODAY)
+    const forecastRowC = buildValidForecastRow(1, TOMORROW)
+    forecastRowC.tempForecast!.value = 27
```

Comment:
Any reason why we're manually setting the tempForecast here?

Reply:
Yes, I want to change the temp in order to force the indices to change for the last day. I'll add a comment!